Upstream: https://github.com/coccinelle/coccinelle/pull/39

From 33ea9df845d11f89b5f1bad65084034613db793a Mon Sep 17 00:00:00 2001
From: Nikolay Orlyuk <virkony@gmail.com>
Date: Sat, 2 May 2015 01:02:47 +0300
Subject: [PATCH] autotools: use libdir instead of datadir

According to semantics used in autotools @datadir@ is a special place
for architecture-independent read-only data files.
Though under COCCINELLE_HOME we have some files that suite this
criteria we also put there ocaml compiled files and python modules.
Most of linux distros use libdir for storing such kind of content.
Example is python modules that usually stored under
/usr/lib/python3.4.
Let's use same approach and migrate default value for COCCINELLE_HOME
to @libdir@/coccinelle from @prefix@/share/coccinelle.
---
 configure.ac                  | 4 ++--
 docs/spatch.1.in              | 4 ++--
 globals/config.ml.in          | 2 +-
 release.nix                   | 6 +++---
 scripts/spatch.sh.in          | 2 +-
 tools/sgen/scripts/sgen.sh.in | 2 +-
 6 files changed, 10 insertions(+), 10 deletions(-)

diff --git a/configure.ac b/configure.ac
index 78323fa..ae89726 100644
--- a/configure.ac
+++ b/configure.ac
@@ -391,7 +391,7 @@ dnl  backwards compatibility
 dnl
 
 dnl  defines some variables needed by the Make infrastructure
-AC_SUBST(SHAREDIR,["${prefix}/share/coccinelle"])
+AC_SUBST(SHAREDIR,["${libdir}/coccinelle"])
 
 AS_IF([test "x$enable_python" != xno],
 [dnl
@@ -612,7 +612,7 @@ You may need to set some environment variables so that the python
 libraries are found. The frontend 'spatch' sets the following variables:
 
   \$ export LD_LIBRARY_PATH=\$LD_LIBRARY_PATH:$prefix/lib
-  \$ export PYTHONPATH=\$PYTHONPATH:$prefix/share/coccinelle/python
+  \$ export PYTHONPATH=\$PYTHONPATH:$libdir/coccinelle/python
 
 EOF
 fi
diff --git a/docs/spatch.1.in b/docs/spatch.1.in
index ce5b998..88eb593 100644
--- a/docs/spatch.1.in
+++ b/docs/spatch.1.in
@@ -165,13 +165,13 @@ set number of diff context lines
 report partial matches of the SP on the C file
 .TP
 \fB\-\-iso\-file\fR
-<file> (default=/usr/local/share/coccinelle/standard.iso)
+<file> (default=/usr/local/lib/coccinelle/standard.iso)
 .TP
 \fB\-\-macro\-file\fR
 <file>
 .TP
 \fB\-\-macro\-file\-builtins\fR
-<file> (default=/usr/local/share/coccinelle/standard.h)
+<file> (default=/usr/local/lib/coccinelle/standard.h)
 .TP
 \fB\-\-recursive\-includes\fR
 causes all available include files, both those included in the C file(s) and those included in header files, to be used
diff --git a/globals/config.ml.in b/globals/config.ml.in
index 31004b8..5d547ff 100644
--- a/globals/config.ml.in
+++ b/globals/config.ml.in
@@ -2,7 +2,7 @@ let version = "@COCCI_VERSION@"
 
 let path =
   try (Sys.getenv "COCCINELLE_HOME")
-  with Not_found->"@prefix@/share/coccinelle"
+  with Not_found->"@libdir@/coccinelle"
 
 let std_iso = ref (Filename.concat path "standard.iso")
 let std_h   = ref (Filename.concat path "standard.h")
diff --git a/release.nix b/release.nix
index 1e39b52..91cbecf 100644
--- a/release.nix
+++ b/release.nix
@@ -481,11 +481,11 @@ let
 
 	# initialize essential environment variables
         # for the makefile
-	export COCCINELLE_HOME=${coccinelle}/share/coccinelle
+	export COCCINELLE_HOME=${coccinelle}/lib/coccinelle
         export COCCIDIR=$TMPDIR
         export SPATCH=${coccinelle}/bin/spatch.opt
-        export ISO=${coccinelle}/share/coccinelle/standard.iso
-        export DEFS=${coccinelle}/share/coccinelle/standard.h
+        export ISO=${coccinelle}/lib/coccinelle/standard.iso
+        export DEFS=${coccinelle}/lib/coccinelle/standard.h
 
 	# generate the test outcomes using a parallel build
         make -e all -j$NIX_BUILD_CORES -l$NIX_BUILD_CORES
diff --git a/scripts/spatch.sh.in b/scripts/spatch.sh.in
index a555352..c5bed3b 100644
--- a/scripts/spatch.sh.in
+++ b/scripts/spatch.sh.in
@@ -4,8 +4,8 @@
 # it sets the python variables (if relevant)
 # and COCCINELLE_HOME
 
-SHAREDIR="@prefix@/share/coccinelle"
 LIBDIR="@prefix@/lib"
+SHAREDIR="${LIBDIR}/coccinelle"
 
 if test ! -d "$SHAREDIR"; then
     SHAREDIR="$(pwd)"
diff --git a/tools/sgen/scripts/sgen.sh.in b/tools/sgen/scripts/sgen.sh.in
index ec95d17..65d31be 100755
--- a/tools/sgen/scripts/sgen.sh.in
+++ b/tools/sgen/scripts/sgen.sh.in
@@ -2,6 +2,6 @@
 
 PREFIX=replace_me
 TARGET=sgen
-SHAREDIR=""${PREFIX}/share/coccinelle/sgen""
+SHAREDIR=""${PREFIX}/lib/coccinelle/sgen""
 
 exec ""${SHAREDIR}/${TARGET}"" "$@"
-- 
2.3.3

